maintainers/scripts/pluginupdate.py

```diff
@@ -753,15 +845,26 @@ def rewrite_input(
             deprecations = json.load(f)
         # TODO parallelize this step
         for pdesc, new_repo in redirects.items():
-            log.info("Rewriting input file %s", input_file)
+            log.info("Resolving deprecated plugin %s -> %s", pdesc.name, new_repo.name)
             new_pdesc = PluginDesc(new_repo, pdesc.branch, pdesc.alias)
+
             old_plugin, _ = prefetch_plugin(pdesc)
             new_plugin, _ = prefetch_plugin(new_pdesc)
+
             if old_plugin.normalized_name != new_plugin.normalized_name:
                 deprecations[old_plugin.normalized_name] = {
                     "new": new_plugin.normalized_name,
                     "date": cur_date_iso,
                 }
+
+            # remove plugin from index file, so we won't add it
+            # to deprecations again
+            for i, plugin in enumerate(plugins):
+                if plugin.name == pdesc.name:
```

Comment:
As you could see at the description of [third commit](https://github.com/NixOS/nixpkgs/pull/336137/commits/9c05ff4ec3e4ec8e909786dba5af4a99d8f14bef), someone just forgot about deleting plugins from index for years!

Reply:
Time for some cleaning 🧹